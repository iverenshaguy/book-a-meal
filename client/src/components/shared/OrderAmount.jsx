import React from 'react';
import PropTypes from 'prop-types';
import { orderMealPropTypes } from '../../helpers/proptypes';
import calculateCashEarnedFromOrder from '../../helpers/calculateCashEarnedFromOrder';

/**
 * @exports
 * @function OrderAmount
 * @param {string} type
 * @param {array} meals
 * @returns {JSX} OrderAmount
 */
const OrderAmount = ({ type, meals }) => (
  <div className={`order-amount ${type === 'admin' && 'admin-order-total'}`}>
    <div>
      <p>Total</p>
      <h2>
â‚¦
        {calculateCashEarnedFromOrder(meals)}
      </h2>
    </div>
  </div>
);

OrderAmount.propTypes = {
  type: PropTypes.string,
  meals: PropTypes.arrayOf(orderMealPropTypes).isRequired
};

OrderAmount.defaultProps = {
  type: 'user'
};

export default OrderAmount;
