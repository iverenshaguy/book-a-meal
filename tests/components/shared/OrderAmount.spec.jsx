import React from 'react';
import OrderAmount from '../../../src/components/shared/OrderAmount';
import { caterersOrdersObj } from '../../setup/mockData';


describe('OrderAmount', () => {
  it('should render OrderAmount component correctly', () => {
    const shallowWrapper = shallow(<OrderAmount type="admin" meals={caterersOrdersObj.orders[0].meals} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
});
