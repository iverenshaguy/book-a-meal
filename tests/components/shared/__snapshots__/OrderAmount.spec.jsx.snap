// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderAmount should render OrderAmount component correctly 1`] = `
<div
  className="order-amount admin-order-total"
>
  <div>
    <p>
      Total
    </p>
    <h2>
      â‚¦
      1200
    </h2>
  </div>
</div>
`;
